### DISCUSSION TITLE
Use Early Stopping to Halt the Training of Neural Networks At the Right Time

### DISCUSSION THREAD

**Discussion by wjunneng** [ourownstory/neural_prophet#760] [Feature Requests & Feedback]
I have training and validation sets, can I add an early stopping strategy to avoid overfitting.

**Comment by wjunneng** [comment]
If I specify "epochs" and don't specify "batch_size", the model won't overfit。

**Comment by ourownstory** [comment]
Hi @wjunneng There is an old draft PR for this up: #289
If you like, you could bring it up to speed with the current state of the project and contribute the feature!
